source/kernel/bank2/rw.mac

```diff
@@ -153,6 +153,12 @@ abs_common_doerr:
 		jr	try_rw_val
 
 abs_rw_dorw:
+		push de		;If writing, reset number of next cluster to allocate
+		bit	RF_READ,(iy+@RW_FLAGS##) ;just in case we're messing up with FAT
+		ld de,1
+		pcall z,ACLU_SET
+		pop de

```

Comment:
If you can spare 9 bytes, I would suggest:
```
    bit RF_READ,(iy+@RW_FLAGS##) ;just in case we're messing up with FAT
    pcall z,ACLU_RESET ;If writing, reset number of next cluster to allocate
```

And then implement `ACLU_RESET` as:
```
	PROC	ACLU_RESET
;
;    Resets the UD_ACLU field of a unit descriptor (sets to -1)
;
; Entry:   HL = Address of unit descriptor
; Returns:     None
; Corrupts:    F
;
	push	hl
	push    de
	ld	de, UD_ACLU##
	add	hl,de
	ld	(hl),0ffh
	inc	hl
	ld	(hl),0ffh
	pop     de
	pop	hl
	ret
```

If I'm not mistaken, this is 28 t-states faster if the branch is taken (149 vs 177 t-states) and 34 t-states faster otherwise (33 vs 67 t-states)

Reply:
Thanks for the suggestion but I think I'll leave this one unchanged. ROM space is becoming scarce so the smaller the code the better, and resetting `UD_ACLU` is something that won't happen often, so it's a process that doesn't need to be heavily optimized.